manifests/vcsscm/git.pp

```diff
@@ -49,30 +49,61 @@
       }
 
       if $gitconfig {
-        file { '/etc/gitconfig':
-          ensure => file,
-          owner  => 'root',
-          group  => 'root',
-          mode   => '0644',
-          source => 'puppet:///modules/software/vcsscm/git/system-gitconfig',
+
+        if $gitconfig =~ Boolean {
+          $gitconfig = {
```

Comment:
I don't think you can reassign variables in puppet... Unless they added that in puppet 4?
You commonly see $_variablename.